src/Components/test/E2ETest/ServerExecutionTests/InteropReliabilityTests.cs

```diff
@@ -308,15 +306,15 @@ public async Task JSInteropCompletionSuccess()
             Assert.NotEqual(default, call);
 
             var id = call.AsyncHandle;
-            await Client.HubConnection.InvokeAsync(
+            await Client.ExpectRenderBatch(() => Client.HubConnection.InvokeAsync(
                 "EndInvokeJSFromDotNet",
-                id++,
+                id,
```

Comment:
This test wasn't flaky, but I think it has always been giving false positives because of this line.

It wasn't completing the same JS interop call that it was triggering, so when it checks "there were no errors", it's not really proving anything. Now it does complete the correct interop call, and explicitly checks for a success message.